loopdb/sql_store.go

```diff
@@ -630,6 +752,170 @@ func (s *BaseDB) convertLoopInRow(row sqlc.GetLoopInSwapsRow,
+	}
+
+	if row.BatchTxID.Valid {
+		err := chainhash.Decode(&batch.BatchTxid, row.BatchTxID.String)
+		if err != nil {
+			return batch, err
+		}
+	}
+
+	batch.BatchPkScript = row.BatchPkScript
+
+	if row.LastRbfHeight.Valid {
+		batch.LastRbfHeight = row.LastRbfHeight.Int32
+	}
+
+	if row.LastRbfSatPerKw.Valid {
+		batch.LastRbfSatPerKw = row.LastRbfSatPerKw.Int32
+	}
+
+	batch.MaxTimeoutDistance = row.MaxTimeoutDistance
+
+	return batch, nil
+}
+
+// BatchToUpsertArgs converts a Batch struct to the arguments needed to insert
+// it into the database.
+func (s *BaseDB) BatchToInsertArgs(
+	batch Batch) sqlc.InsertBatchParams {
+
+	args := sqlc.InsertBatchParams{
+		State: batch.State,
+		BatchTxID: sql.NullString{
+			Valid:  true,
+			String: batch.BatchTxid.String(),
+		},
+		BatchPkScript: batch.BatchPkScript,
+		LastRbfHeight: sql.NullInt32{
+			Valid: true,
+			Int32: batch.LastRbfHeight,
+		},
+		LastRbfSatPerKw: sql.NullInt32{
+			Valid: true,
+			Int32: batch.LastRbfSatPerKw,
+		},
+		MaxTimeoutDistance: batch.MaxTimeoutDistance,
+	}
+
+	return args
+}
+
+// BatchToUpsertArgs converts a Batch struct to the arguments needed to insert
+// it into the database.
+func (s *BaseDB) BatchToUpdateArgs(
+	batch Batch) sqlc.UpdateBatchParams {
+
+	args := sqlc.UpdateBatchParams{
+		ID:    batch.ID,
+		State: batch.State,
+		BatchTxID: sql.NullString{
+			Valid:  true,
+			String: batch.BatchTxid.String(),
+		},
+		BatchPkScript: batch.BatchPkScript,
+		LastRbfHeight: sql.NullInt32{
+			Valid: true,
+			Int32: batch.LastRbfHeight,
+		},
+		LastRbfSatPerKw: sql.NullInt32{
+			Valid: true,
+			Int32: batch.LastRbfSatPerKw,
+		},
+	}
+
+	return args
+}
+
+// ConvertSweepRow converts a sweep row from db to a sweep struct.
+func (s *BaseDB) ConvertSweepRow(ctx context.Context,
+	row sqlc.GetBatchSweepsRow) (Sweep, error) {
+
+	sweep := Sweep{
+		ID:      row.ID,
+		BatchID: row.BatchID,
+		Amount:  btcutil.Amount(row.Amt),
+	}
+
+	swapHash, err := lntypes.MakeHash(row.SwapHash)
+	if err != nil {
+		return sweep, err
+	}
+
+	sweep.SwapHash = swapHash
+
+	hash, err := chainhash.NewHash(row.OutpointTxid)
+	if err != nil {
+		return sweep, err
+	}
+
+	sweep.Outpoint = wire.OutPoint{
+		Hash:  *hash,
+		Index: uint32(row.OutpointIndex),
+	}
+
+	updates, err := s.Queries.GetSwapUpdates(ctx, row.SwapHash)
+	if err != nil {
+		return sweep, err
+	}
+
+	sweep.LoopOut, err = s.convertLoopOutRow(
+		sqlc.GetLoopOutSwapRow{
+			ID:                     row.ID,
+			SwapHash:               row.SwapHash,
+			Preimage:               row.Preimage,
+			InitiationTime:         row.InitiationTime,
+			AmountRequested:        row.AmountRequested,
+			CltvExpiry:             row.CltvExpiry,
+			MaxMinerFee:            row.MaxMinerFee,
+			MaxSwapFee:             row.MaxSwapFee,
+			InitiationHeight:       row.InitiationHeight,
+			ProtocolVersion:        row.ProtocolVersion,
+			Label:                  row.Label,
+			SwapHash_2:             row.SwapHash_2,
```

Comment:
Hmm this is something we never caught, but it seems that `GetLoopOutSwap` redundantly returns swap hash multiple times. Perhaps consider adding a fixup commit for this. (Can also do in follow up PR although it is trivial).